src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1449,6 +1463,29 @@ private void MergeHidingLookupResults(LookupResult resultHiding, LookupResult re
             {
                 resultHiding.MergePrioritized(resultHidden);
             }
+
+            return;
+
+            static bool isInExtensionHierarchy(NamedTypeSymbol hiddenContainer, NamedTypeSymbol hidingContainer, ConsList<TypeSymbol> basesBeingResolved, CSharpCompilation compilation, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+            {
+                if (!hidingContainer.IsExtension)
+                {
+                    return false;
+                }
+
+                var extendedType = hidingContainer.ExtendedTypeNoUseSiteDiagnostics;
+                if (hiddenContainer.Equals(extendedType, TypeCompareKind.ConsiderEverything))
```

Comment:
>TypeCompareKind.ConsiderEverything [](http://example.com/codeflow?start=57&length=34)

Why not ignore everything? Do we have a test demonstrating significance of this option?

Reply:
I don't think the difference is observable. I picked `ConsiderEverything` to match `ConsiderEverything2` that we use in rest of `MergeHidingLookupResults` (inside `IsDerivedType`).